docs/vc-spec.md

```diff
@@ -1,6 +1,163 @@
-# II Verifiable Credential Spec
+# II Verifiable Credential Spec (MVP)
+
+
+## Issuer API
+
+This section describes the (Candid) interface to be implemented by an issuer of verifiable credentials on the IC.
+This interface is used by the II-canister during attribute sharing flow (cf. [flow description](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/attribute-sharing.md))
+An example implementation of the interface is given in [demos/vc_issuer](../demos/vc_issuer).
+
+The Candid interface is as follows, and the subsequent sections describe the
+services and the corresponding messages in more detail.
+
+```candid
+type CredentialSpec = record { info : text };
+type GetCredentialRequest = record {
+    signed_id_alias : SignedIdAlias;
+    vc_jwt : opt text;
+    credential_spec : CredentialSpec;
+};
+type GetCredentialResponse = variant {
+    Ok : IssuedCredentialData;
+    Err : IssueCredentialError
+};
+type Icrc21ConsentInfo = record { consent_message : text; language : text };
+type Icrc21ConsentMessageRequest = record {
+    arg : vec nat8;
+    method : text;
+    preferences : Icrc21ConsentPreferences;
+};
+type Icrc21ConsentMessageResponse = variant {
+    Ok : Icrc21ConsentInfo;
+    Err : Icrc21Error;
+};
+type Icrc21ConsentPreferences = record { language : text };
+type Icrc21Error = variant {
+    GenericError : Icrc21ErrorInfo;
+    MalformedCall : Icrc21ErrorInfo;
+    NotSupported : Icrc21ErrorInfo;
+    Forbidden : Icrc21ErrorInfo;
+};
+type Icrc21ErrorInfo = record { description : text; error_code : nat64 };
+type IssueCredentialError = variant {
```

Comment:
Do we have a way for the II front-end to handle the UX if the user does not fulfill the prerequisites for the credential.

I assume this is what the `UnauthorizedSubject` error is for? If so, it would be nice if the issuer could attach a human readable message to be shown to the user (with some explanation).

Example: "You are not eligible for this credential because your 8 year neuron has insufficient stake".

Although, in that case we would need to consider user-preferences (similar to the consent message). 🙃

Reply:
As discussed, I added a text message to all the variants.